tests/test_stp_global.rb

```diff
@@ -0,0 +1,533 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/stp_global'
+
+# TestX__CLASS_NAME__X - Minitest for X__CLASS_NAME__X node utility class
+class TestStpGlobal < CiscoTestCase
+  # TESTS
+
+  DEFAULT_NAME = 'default'
+
+  def setup
+    super
+    config 'no spanning-tree mode'
+    config 'system bridge-domain none' if n7k_platform?
+  end
+
+  def teardown
+    config 'no spanning-tree mode'
+    config 'system bridge-domain none' if n7k_platform?
+    super
+  end
+
+  def n7k_platform?
+    /N7/ =~ node.product_id
+  end
+
+  def n9k_platform?
+    /N(3|9)/ =~ node.product_id
+  end
+
+  def n6k_platform?
+    /N(5|6)/ =~ node.product_id
+  end
+
+  def create_stp_global(name=DEFAULT_NAME)
+    StpGlobal.new(name)
+  end
+
+  def test_bd_forward_time_change
+    skip('Platform does not support this property') if n6k_platform? ||
+                                                       n9k_platform?
+    @global = create_stp_global
+    config 'system bridge-domain all'
+    bdft = [['2-4,6,8-12', '4'], %w(14 30)]
+    @global.bd_forward_time = bdft
+    assert_equal(bdft, @global.bd_forward_time)
+    bdft = [['2-42', '26'], ['83-92,1000-2300', '20'],
+            ['3000-3960', '16']]
+    @global.bd_forward_time = bdft
+    assert_equal(bdft, @global.bd_forward_time)
+    @global.bd_forward_time = @global.default_bd_forward_time
+    assert_equal(@global.default_bd_forward_time,
+                 @global.bd_forward_time)
+  end
+
+  def test_bd_hello_time_change
+    skip('Platform does not support this property') if n6k_platform? ||
+                                                       n9k_platform?
+    @global = create_stp_global
+    config 'system bridge-domain all'
+    bdft = [['2-4,6,8-12', '1'], %w(14 10)]
+    @global.bd_hello_time = bdft
+    assert_equal(bdft, @global.bd_hello_time)
+    bdft = [['2-42', '6'], ['83-92,1000-2300', '9'],
+            ['3000-3960', '3']]
+    @global.bd_hello_time = bdft
+    assert_equal(bdft, @global.bd_hello_time)
+    @global.bd_hello_time = @global.default_bd_hello_time
+    assert_equal(@global.default_bd_hello_time,
+                 @global.bd_hello_time)
+  end
+
+  def test_bd_max_age_change
+    skip('Platform does not support this property') if n6k_platform? ||
+                                                       n9k_platform?
+    @global = create_stp_global
+    config 'system bridge-domain all'
+    bdft = [['2-4,6,8-12', '10'], %w(14 40)]
+    @global.bd_max_age = bdft
+    assert_equal(bdft, @global.bd_max_age)
+    bdft = [['2-42', '6'], ['83-92,1000-2300', '29'],
+            ['3000-3960', '13']]
+    @global.bd_max_age = bdft
+    assert_equal(bdft, @global.bd_max_age)
+    @global.bd_max_age = @global.default_bd_max_age
+    assert_equal(@global.default_bd_max_age,
+                 @global.bd_max_age)
+  end
+
+  def test_bd_priorities_change
```

Comment:
Took care of this as well. The issue is that when we have a mix of vlans and bd, the reset must be done only on the bds and not on the vlans. So added a new method to get the bd range exclusive of the vlans and resetting them if necessary.